packages/contracts/contracts/generator/GenArt721GeneratorV0.sol

```diff
@@ -0,0 +1,331 @@
+    using Bytes32Strings for string;
+
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    DependencyRegistryV0 public dependencyRegistry;
+    IScriptyBuilderV2 public scriptyBuilder;
+    address public ethFS;
+
+    function _onlySupportedCoreContract(
+        address coreContractAddress
+    ) internal view {
+        require(
+            dependencyRegistry.isSupportedCoreContract(coreContractAddress),
+            "Unsupported core contract"
+        );
+    }
+
+    /**
+     * @notice Constructor for the GenArt721GeneratorV0 contract.
+     * @param _dependencyRegistry The address of the DependencyRegistry
+     * contract to be used for retrieving dependency scripts.
+     * @param _scriptyBuilder The address of the ScriptyBuilderV2 contract
+     * to be used for generating the HTML for tokens.
+     * @param _ethFS The address of the EthFSFileStorage contract to retrieve
+     * the gunzipScripts-0.0.1.js script from used to gunzip the dependency
+     * scripts.
+     */
+    constructor(
+        address _dependencyRegistry,
+        address _scriptyBuilder,
+        address _ethFS
+    ) {
+        dependencyRegistry = DependencyRegistryV0(_dependencyRegistry);
+        scriptyBuilder = IScriptyBuilderV2(_scriptyBuilder);
+        ethFS = _ethFS;
+    }
+
+    /**
+     * @notice Get the dependency script for a given dependency name and version.
+     * @param dependencyNameAndVersion The name and version of the dependency
+     * to retrieve the script for.
+     * @return The dependency script stiched together from all of the script chunks.
+     * @dev This function will revert if the dependency name and version is not
+     * supported by the DependencyRegistry or if the script storage version is
+     * not supported.
+     */
+    function getDependencyScript(
+        bytes32 dependencyNameAndVersion
+    ) external view returns (bytes memory) {
+        uint256 scriptCount = dependencyRegistry.getDependencyScriptCount(
+            dependencyNameAndVersion
+        );
+
+        if (scriptCount == 0) {
+            return "";
+        }
+
+        address[] memory scriptBytecodeAddresses = new address[](scriptCount);
+
+        for (uint256 i = 0; i < scriptCount; i++) {
+            scriptBytecodeAddresses[i] = dependencyRegistry
+                .getDependencyScriptBytecodeAddress(
+                    dependencyNameAndVersion,
+                    i
+                );
+        }
+
+        bytes32 storageVersion = BytecodeStorageReader
+            .getLibraryVersionForBytecode(scriptBytecodeAddresses[0]);
+
+        uint256 offset;
+        if (storageVersion == BytecodeStorageReader.V0_VERSION_STRING) {
+            offset = 104;
+        } else if (storageVersion == BytecodeStorageReader.V1_VERSION_STRING) {
+            offset = 65;
+        } else {
+            revert("Unsupported storage version");
+        }
+
+        return AddressChunks.mergeChunks(scriptBytecodeAddresses, offset);
```

Comment:
I think the reason we get the data this way instead of using `BytecodeStorageReader.readFromBytecode` and copying strings around in memory is for gas efficiency, right? I think probably worth a small dev comment about why this implementation was selected.